cosmwasm/its.ts

```diff
@@ -0,0 +1,186 @@
+import { CosmWasmClient } from '@cosmjs/cosmwasm-stargate';
+import { Argument, Command, Option } from 'commander';
+import { Contract, constants, getDefaultProvider } from 'ethers';
+
+import { addEnvOption, tokenManagerTypes, validateParameters } from '../common';
+import { printError, printInfo } from '../common';
+import { ConfigManager } from '../common/config';
+import { getContractJSON } from '../evm/utils';
+import { ClientManager, mainProcessor, mainQueryProcessor } from './processor';
+
+const IInterchainToken = getContractJSON('IInterchainToken');
+
+export type TokenData = {
+    tokenId: string;
+    originChain: string;
+    decimals: number;
+    chainName: string;
+};
+
+export async function registerToken(
+    config: ConfigManager,
+    interchainTokenServiceAddress: string,
+    client: ClientManager,
+    tokenData: TokenData,
+    dryRun: boolean,
+) {
+    const alreadyRegistered = await checkSingleTokenRegistration(
+        config,
+        client,
+        interchainTokenServiceAddress,
+        tokenData.tokenId,
+        tokenData.chainName,
+    );
+    if (alreadyRegistered) {
+        printInfo(`Token ${tokenData.tokenId} on ${tokenData.chainName} is already registered`);
+        return;
+    }
+
+    const msg = {
+        register_p2p_token_instance: {
+            chain: config.getChainConfig(tokenData.chainName).axelarId,
+            token_id: formatTokenId(tokenData.tokenId),
+            origin_chain: config.getChainConfig(tokenData.originChain).axelarId,
+            decimals: tokenData.decimals,
+            supply: 'untracked',
+        },
+    };
+
+    const [account] = client.accounts;
+    printInfo('Registering token ', JSON.stringify(msg.register_p2p_token_instance));
+
+    if (!dryRun) {
+        await client.execute(account.address, interchainTokenServiceAddress, msg, 'auto');
+    }
+}
+
+export async function checkSingleTokenRegistration(
```

Comment:
```suggestion
export async function checkTokenRegistrationByChain(
```
specify the category that single refers to instead